Composer/packages/client/src/shell/useShell.ts

```diff
@@ -125,32 +105,44 @@ export function useShell(source: EventSource): { api: ShellApi; data: ShellData
     }
   }, [schemas, projectId]);
 
+  dialogMapRef.current = dialogsMap;
+
   const api: ShellApi = {
+    getDialog: (dialogId: string) => {
+      return dialogMapRef.current[dialogId];
```

Comment:
@lei9444 there is a bug here related to the usage of `ref.current`
If I moved some actions to a new dialog, composer data will be updated but the actual file won't be updated until next time I edit the file.

Reply:
@yeze322 I have fix this bug in #2742 , It's not related to the ref. It's a bug in file persistence sync logic

Reply:
refs #2742